sentry-android-core/src/main/java/io/sentry/android/core/AnrV2Integration.java

```diff
@@ -240,9 +256,24 @@ private void reportAsSentryEvent(
       final Hint hint = HintUtils.createWithTypeCheckHint(anrHint);
 
       final SentryEvent event = new SentryEvent();
-      event.setThreads(threads);
-      event.setTimestamp(DateUtils.getDateTime(anrTimestamp));
+      if (result.type == ParseResult.Type.ERROR) {
+        final Message sentryMessage = new Message();
+        sentryMessage.setFormatted(
+            "Sentry Android SDK failed to parse system thread dump for "
+                + "this ANR. We recommend enabling [SentryOptions.isAttachAnrThreadDump] option "
+                + "to attach the thread dump as plain text and report this issue on GitHub.");
+        event.setMessage(sentryMessage);
+      } else if (result.type == ParseResult.Type.DUMP) {
+        event.setThreads(result.threads);
+      }
       event.setLevel(SentryLevel.FATAL);
+      event.setTimestamp(DateUtils.getDateTime(anrTimestamp));
+
+      if (options.isAttachAnrThreadDump()) {
```

Comment:
So the dump will always get attached if enabled right? I wondering if it makes sense to attach it only when `result.type == ParseResult.Type.ERROR`

Reply:
Yeah, I thought about this, but I think it could be still valuable even if our parsing was successful, because the complete thread dump has a bit more information than just threads (also available memory size, etc etc). So I'd just leave this up to the user to decide